### PR TITLE
Expose time since published for videos as item property

### PR DESCRIPTION
**Pull request by pannal**
This PR is in reference to #329 and somewhat a proof-of-concept. 
- fixes a couple of issues inside the translation files
- adds item properties `PublishedSince` (dynamic localized short description for publishedAt) and `PublishedLocal` (the full datetime object), both adjusted to the local timezone

`PublishedSince` can be:
- Just now
- A minute ago
- Recently
- An hour ago
- Two hours ago
- Three hours ago
- Yesterday at TIME
- Two days ago
- Today at TIME
- The date and time of the publication

All of those strings are localized.

### CODE REVIEW

**Comment by anxdpanic** [comment]
Thanks for the PR, a few things need minor changes before merge.
The first one, can you change the usage of " to ' in .py files so it matches current code.
I'll comment on areas to be more specific with the rest (nothing major)

File: resources/lib/youtube_plugin/kodion/impl/xbmc/xbmc_items.py [hunk 1]
```diff
@@ -104,6 +105,12 @@ def to_video_item(context, video_item):
 
     item.setProperty(u'IsPlayable', u'true')
 
+    publishedAt = video_item.get_aired_utc()
+    if publishedAt:
+        local_dt = item_utils.utc_to_local(publishedAt)
```
Comment (anxdpanic) [inline resources/lib/youtube_plugin/kodion/impl/xbmc/xbmc_items.py]: can you move utc_to_local, datetime_to_since to kodion/datetime_parser and then call by utils.datetime_parser.utc_to_local

File: resources/lib/youtube_plugin/kodion/impl/xbmc/xbmc_items.py [hunk 2]
```diff
@@ -104,6 +105,12 @@ def to_video_item(context, video_item):
...
+        item.setProperty(u"PublishedSince", unicode(item_utils.datetime_to_since(local_dt, context)))
```
Comment (anxdpanic) [inline resources/lib/youtube_plugin/kodion/impl/xbmc/xbmc_items.py]: change unicode to utils.to_unicode for py3

File: resources/lib/youtube_plugin/kodion/impl/xbmc/xbmc_items.py [hunk 3]
```diff
@@ -104,6 +105,12 @@ def to_video_item(context, video_item):
...
+        item.setProperty(u"PublishedLocal", unicode(local_dt))
```
Comment (anxdpanic) [inline resources/lib/youtube_plugin/kodion/impl/xbmc/xbmc_items.py]: same here, to_unicode

File: resources/lib/youtube_plugin/youtube/helper/utils.py [hunk 4]
```diff
@@ -260,6 +261,12 @@ def update_video_infos(provider, context, video_id_dict, playlist_item_id_dict=N
         # date time
         if not datetime and 'publishedAt' in snippet:
             datetime = utils.datetime_parser.parse(snippet['publishedAt'])
+            try:
+                parsed_dt = dt.datetime.strptime(snippet['publishedAt'], "%Y-%m-%dT%H:%M:%S.%fZ")
+            except TypeError:
+                # see https://forum.kodi.tv/showthread.php?tid=112916
+                parsed_dt = dt.datetime(*time.strptime(snippet['publishedAt'], "%Y-%m-%dT%H:%M:%S.%fZ")[:6])
```
Comment (anxdpanic) [inline resources/lib/youtube_plugin/youtube/helper/utils.py]: can you move this to a function in kodion/datetime_parser and access by utils.datetime_parser
also take a look at https://github.com/MrSprigster/Twitch-on-Kodi/blob/master/resources/lib/addon/utils.py#L259-L265

**Comment by pannal** [comment]
Normalized the strings.

As a side note (I've commented on that in the linked ticket): You seem to be using the UTC datetime which YouTube sets, bare-bones without converting it to the local timezone. I've added a conversion function for that, would it make sense to use that in general?

**Comment by anxdpanic** [comment]
I'd save that for another PR, but normalizing utc to local would be welcome

**Comment by pannal** [comment]
OK, fixed everything you requested. Any idea for ```
item.setProperty(u'PublishedLocal',
                         utils.to_unicode(unicode(local_dt)))
```?

`utils.to_unicode` can only accept strings, `local_dt` is a `datetime` object. Using `str` instead would break on non-ascii.

**Comment by anxdpanic** [comment]
Thank you 👍

**Comment by anxdpanic** [comment]
Closes #329

**Pull request merged**